etc/clang-flags.cmake

```diff
@@ -1,13 +1,30 @@
+# cmake-format: off
+# etc/clang-flags.cmake -*-makefile-*-
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+# cmake-format: on
+
 include_guard(GLOBAL)
 
-set(CMAKE_CXX_STANDARD 23)
+if(NOT "${CMAKE_CXX_STANDARD}")
```

Comment:
I'll work on fixing this up, but Standard level is one of the things that needs to be a fixed part of a toolchain so that ABI is consistent across components built with the same toolchain, it's not just about which compiler. 
We're building everything in tree and not depending on anything that has polyfills so it's not going to cause a problem for optional26, but it does in general. In particular once modules are in play.